pkg/scannerv4/mappers/mappers.go

```diff
@@ -671,137 +643,86 @@ func pkgFixedBy(enrichments map[string][]json.RawMessage) (map[string]string, er
 
-	// NVD CVSS metrics
-	nvdSev, _ := nvdSeverityAndScores(vuln, nvdVuln)
-	nvdCVSS, nvdErr := toCVSSMetric(nvdSev)
-	if nvdErr == nil {
-		cvssMetrics = append(cvssMetrics, nvdCVSS)
-	}
-
-	// Undecided CVSS metrics
-	// ignore error since invalid sev values will trigger error in toCVSSMetric
-	undecidedSev, _ := cvssVector(vuln.Severity)
-	var undecidedCVSS *v4.VulnerabilityReport_Vulnerability_CVSS
-	var undecidedErr error
-	undecidedSev.url = vuln.Links
+	var preferredCVSS *v4.VulnerabilityReport_Vulnerability_CVSS
+	var preferredErr error
 	switch {
-	case strings.HasPrefix(vuln.Updater, osvUpdaterPrefix):
-		undecidedSev.source = v4.VulnerabilityReport_Vulnerability_CVSS_SOURCE_OSV
-		undecidedCVSS, undecidedErr = toCVSSMetric(undecidedSev)
+	case rhelUpdaterPattern.MatchString(vuln.Updater):
+		preferredCVSS, preferredErr = rhelCVSS(vuln)
+	case strings.HasPrefix(vuln.Updater, osvUpdaterPrefix) && !isOSVDBSpecificSeverity(vuln.Severity):
+		preferredCVSS, preferredErr = vulnCVSS(vuln, v4.VulnerabilityReport_Vulnerability_CVSS_SOURCE_OSV)
 	case strings.EqualFold(vuln.Updater, constants.ManualUpdaterName):
-		undecidedSev.source = v4.VulnerabilityReport_Vulnerability_CVSS_SOURCE_UNKNOWN
-		undecidedCVSS, undecidedErr = toCVSSMetric(undecidedSev)
+		preferredCVSS, preferredErr = vulnCVSS(vuln, sourceFromLinks(vuln.Links))
 	}
-
-	// Append undecided CVSS if no error
-	if undecidedErr == nil && undecidedCVSS != nil {
-		cvssMetrics = append(cvssMetrics, undecidedCVSS)
+	if preferredCVSS != nil {
+		metrics = append(metrics, preferredCVSS)
 	}
 
-	// Decide primary severity based on the updater
-	switch {
-	case rhelUpdaterPattern.MatchString(vuln.Updater):
-		if rhelErr != nil {
-			zlog.Debug(ctx).
-				Err(rhelErr).
-				Str("severity", vuln.Severity).
-				Msg("failed to get CVSS vector")
-			return cvssMetrics, nil, rhelSev.severity, rhelErr
-		}
-		return cvssMetrics, rhelCVSS, rhelSev.severity, nil
-
-	case strings.HasPrefix(vuln.Updater, osvUpdaterPrefix):
-		if undecidedErr != nil || isOSVDBSpecificSeverity(vuln.Severity) {
-			if undecidedErr == nil {
-				undecidedErr = errors.New("OSVDB-specific severity is not valid for CVSS")
-			}
-			zlog.Debug(ctx).
-				Err(undecidedErr).
-				Str("severity", vuln.Severity).
-				Msg("failed to get CVSS vector, falling back to NVD")
-			break
-		}
-		return cvssMetrics, undecidedCVSS, undecidedSev.severity, nil
-
-	case strings.EqualFold(vuln.Updater, constants.ManualUpdaterName):
-		if undecidedErr != nil {
-			zlog.Debug(ctx).
-				Err(undecidedErr).
-				Str("severity", vuln.Severity).
-				Msg("failed to get CVSS vector, falling back to NVD")
-			break
+	var nvdErr error
+	// Manually added vulnerabilities may have its data sourced from NVD.
+	// In that scenario, there is no need to add yet another NVD entry,
+	// especially since there is a reason the manual entry exists in the first place.
+	if preferredCVSS.GetSource() != v4.VulnerabilityReport_Vulnerability_CVSS_SOURCE_NVD {
```

Comment:
Not sure if I am understanding correctly. If preferredCVSS is from OSV, the CVSS metrics will have OSV, NVD. Rhel score won't be there because the vuln data is not from rhel updater any way?

Reply:
Right. It's pretty much:

1. RHEL, NVD
2. OSV, NVD
3. Manual, NVD
4. Manual (when the source is already from NVD, which all current manual vulns are)
5. NVD (when the vuln is not from RHEL, OSV, nor manual updater)

Does this make sense? Is this consistent with what we did before?